storage_controller/src/persistence.rs

```diff
@@ -315,6 +316,31 @@ impl Persistence {
         }
     }
 
+    pub(crate) async fn set_node_availability_zone_id(
```

Comment:
Thinking ahead: I guess we can probably retire this fn in a week or two, once all the existing nodes have an AZ and registrations include it?

Reply:
Yeah, the PR for that is drafted [here](https://github.com/neondatabase/neon/pull/8856)